### DISCUSSION TITLE
HardwareTimer cannot accept global variables

### DISCUSSION THREAD

**Discussion by rsc9421** [stm32duino/Arduino_Core_STM32#1620] [General]
Hi, i was trying this example:
https://github.com/stm32duino/STM32Examples/blob/main/examples/Peripherals/HardwareTimer/Timebase_callback_with_parameter/Timebase_callback_with_parameter.ino
but I can't give 2 arguments to the function instead of one, so I decided to use global variables but doesn't work
how can I fix it?

**Comment by ABOSTM** [comment]
HI @rsc9421,
Firstly, use of global variables should work,
So can you please provide the sketch you used ?
And can you please details what you mean by "doesn't work"?

Secondly, an alternative is to use a single parameter, which is a pointeur, and it can be a pointer to a data structure, containing all the data you wanted.

**Reply by fpistm** [reply]
Mainly, is your global variable is declared as volatile?
